core/include/traccc/fitting/kalman_filter/kalman_fitter.hpp

```diff
@@ -244,6 +252,14 @@ class kalman_fitter {
                 detray::navigation::direction::e_backward);
             fitter_state.m_fit_actor_state.backward_mode = true;
 
+            if (propagation._stepping().dir()[0] == 0.f ||
```

Comment:
&& instead of ||? But when do you observe this exactly?

Reply:
Sorry this should read:

```cpp
if (vector::theta(propagation._stepping().dir()) == 0.f) {
```

Reason for this is that tracks with theta 0 will produce division by zero errors in detray:

https://github.com/acts-project/detray/blob/2d6be9366462218e52723a2ac90dcd39a0e9812c/core/include/detray/propagator/detail/jacobian_engine.hpp#L60

Reply:
I think this should include $\pi$:
```
if (vector::theta(propagation._stepping().dir()) == 0.f || vector::theta(propagation._stepping().dir()) == pi) { 
```
But isn't the following better because trigonometric used in `vector::theta` might be expensive?:
```
if (propagation._stepping().dir()[0] ==0 && propagation._stepping().dir()[1] ==0){
}
```

Reply:
Updated.